src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,356 @@
+                            {
+                                if (cancellationToken.IsCancellationRequested)
+                                    return toArchive;
+
+                                var clonedId = multiIt.CurrentKey.Clone(options.Context.Transaction.InnerTransaction.Allocator);
+
+                                using (var document = _database.DocumentsStorage.Get(options.Context, clonedId,
+                                           DocumentFields.Id | DocumentFields.Data | DocumentFields.ChangeVector))
+                                {
+                                    if (document is null || document.TryGetMetadata(out var metadata) == false || HasPassed(metadata, options.CurrentTime) == false)
+                                    {
+                                        docsToArchive.Add((clonedId, null));
+                                        continue;
+                                    }
+                                    docsToArchive.Add((clonedId, document.Id));
+                                }
+                            } while (multiIt.MoveNext() && docsToArchive.Count + count < options.AmountToTake);
+                        }
+                    }
+
+                    count += docsToArchive.Count;
+                    if (docsToArchive.Count>0)
+                        toArchive.Add(ticksAsSlice,docsToArchive);
+                    
+                    
+                } while (it.MoveNext() && count < options.AmountToTake);
+
+                return toArchive;
+            }
+        }
+
+        // private (bool AllExpired, string Id) GetConflictedArchival(DocumentsOperationContext context, DateTime currentTime, Slice clonedId)
+        // {
+        //     string id = null;
+        //     var allToArchive = true;
+        //     var conflicts = _database.DocumentsStorage.ConflictsStorage.GetConflictsFor(context, clonedId);
+        //     if (conflicts.Count > 0)
+        //     {
+        //         foreach (var conflict in conflicts)
+        //         {
+        //             using (conflict)
+        //             {
+        //                 id = conflict.Id;
+        //
+        //                 if (HasPassed(conflict.Doc, currentTime))
+        //                     continue;
+        //
+        //                 allToArchive = false;
+        //                 break;
+        //             }
+        //         }
+        //     }
+        //
+        //     return (allToArchive, id);
+        // }
+        
+        private static bool HasPassed(BlittableJsonReaderObject metadata, DateTime currentTime)
+        {
+            if (metadata.TryGet(Constants.Documents.Metadata.Archive, out string archiveDate))
+            {
+                if (DateTime.TryParseExact(archiveDate, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var date))
+                {
+                    if (currentTime >= date.ToUniversalTime())
```

Comment:
We can probably assume that this is universal already and add assert for that, rather do the conversion each call

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/15715/commits/463cca9133448f27a26741c54bd4021b09f92ca3